src/code/templates/fv-egg-game.js

```diff
@@ -822,7 +822,7 @@ export default class FVEggGame extends Component {
                             showHatchAnimation: showUserDrake,
                             clearAnimatedComponents: true });
       this.autoSelectSingletonGametes(nextGametes);
-      if (newTrialInChallenge && onResetGametes) {
+      if ((newChallenge || newTrialInChallenge) && onResetGametes) {
```

Comment:
Nice, this has been an outstanding issue for a while. Should we do this for new levels and missions as well?